exp/controllers/azuremanagedmachinepool_reconciler.go

```diff
@@ -153,7 +153,7 @@ func (s *azureManagedMachinePoolService) Reconcile(ctx context.Context, scope *s
 
 	var providerIDs = make([]string, len(instances))
 	for i := 0; i < len(instances); i++ {
-		providerIDs[i] = azure.ProviderIDPrefix + *instances[i].ID
+		providerIDs[i] = strings.ToLower(azure.ProviderIDPrefix + *instances[i].ID)
```

Comment:
do we want to modify `instances[i].ID` as it's being set by the azure api?

Reply:
I believe the intent is to normalize it as the node ID is in all lowercase #1585 

@alexeldeib does Azure cloud provider make the IDs all lower case intentionally or is that a k8s node thing?

Reply:
I believe it’s cloud provider, I imagine because ARM ignores case sensitivity for those comparisons (I think)

Reply:
I just tried querying using azure cli by passing lowercase id, and it works:
```
±  |master → origin ↑644 {6} ✓| → az resource show --ids /subscriptions/eba22861-4990-4213-ab8e-776393e3696b/resourcegroups/test_group/providers/microsoft.compute/disks/test_disk1_0f55a550a08c4d50a5fe9535a005ad29
{
  "id": "/subscriptions/eba22861-4990-4213-ab8e-776393e3696b/resourceGroups/test_group/providers/Microsoft.Compute/disks/test_disk1_0f55a550a08c4d50a5fe9535a005ad29",
...
```